src/test/java/com/google/sps/AbuseDetectionTest.java

```diff
@@ -0,0 +1,225 @@
+    Assert.assertEquals(false, value);
+  }
+
+  /*
+  * 10 Requests are added. First timestamp: 18:19:20.010. The
+  * Last Timestamp 18:19:20.866. 
+  */
+  @Test
+  public void addingMultipleRequestsGetTrue() {
+    Duration currentDur = Duration.ofSeconds(1);
+    AbuseDetection instance = new AbuseDetection(currentDur, 10);
+    add10Requests(instance);
+    LocalTime inputTime = LocalTime.parse("18:19:21.012");
+    boolean value = instance.addRequest(inputTime);
+    // 10 requests were added since 18:19:20.010, thus the 11th is not 
+    // within a second, so it gets added, which returns true
+    Assert.assertEquals(true, value);
+  }
+
+  /*
+  * 10 Requests are added. First timestamp: 18:19:20.010. The
+  * Last Timestamp 18:19:20.866. 
+  */
+  @Test
+  public void requestOverSecondReturnsFalse() {
+    Duration currentDur = Duration.ofSeconds(1);
+    AbuseDetection instance = new AbuseDetection(currentDur, 10);
+    add10Requests(instance);
+    LocalTime inputTime1 = LocalTime.parse("18:19:21.012");
+    boolean curVal = instance.addRequest(inputTime1);
+    LocalTime otherTime = LocalTime.parse("18:19:21.109");
+    boolean value = instance.addRequest(otherTime);
+    // 10 requests added since 18:19:20.010, 11th request isn't within
+    // second, that gets added. The 12th request is wihin a second, so 
+    // it does not get added, and returns false
+    Assert.assertEquals(false, value);
+  }
+  
+  /*
+  * 10 Requests are added. First timestamp: 18:19:20.010. The
+  * Last Timestamp 18:19:20.866. 
+  */
+  @Test
+  public void requestOverSecondReturnsTrue() {
+    Duration currentDur = Duration.ofSeconds(1);
+    AbuseDetection instance = new AbuseDetection(currentDur, 10);
+    add10Requests(instance);
+    LocalTime inputTime1 = LocalTime.parse("18:19:21.012");
+    boolean curVal = instance.addRequest(inputTime1);
+    LocalTime otherTime = LocalTime.parse("18:19:21.109");
+    boolean valueReturned = instance.addRequest(otherTime);
+    LocalTime timeInputted = LocalTime.parse("18:19:21.416");
+    boolean value = instance.addRequest(timeInputted);
+    // 10 requests added since 18:19:20.010, 11th request isn't within
+    // second, that gets added. The 12th request is wihin a second, so 
+    // it does not get added, and returns false. The 13th request isn't 
+    // within a second so it returns true and gets added.
+    Assert.assertEquals(true, value);
+  }
+
+  /*
+  * 10 Requests are added. First timestamp: 23:59:55.010. The
+  * Last Timestamp 00:00:00.000. 
+  */
+  @Test
+  public void adding11RequestsNearMidnight() {
+    Duration currentDur = Duration.ofSeconds(1);
+    AbuseDetection instance = new AbuseDetection(currentDur, 10);
+    add10RequestsNearMidnight(instance);
+    boolean value = instance.addRequest(LocalTime.parse("00:01:01.020"));
+    // 10 Requests were added since 23:59:55.010, the 11th request is
+    // within a second so it cannot get added, so it gets true
```

Comment:
Could you clarify this comment: "The 11th request is within a second so it cannot get added, so it gets true"

Reply:
On it!